core/src/main/java/org/apache/brooklyn/core/config/internal/AbstractConfigMapImpl.java

```diff
@@ -107,4 +252,227 @@ public int size() {
     public boolean isEmpty() {
         return ownConfig.isEmpty();
     }
+    
+    protected ConfigInheritance getDefaultRuntimeInheritance() {
+        return BasicConfigInheritance.OVERWRITE; 
+    }
+
+    public <T> ReferenceWithError<ConfigValueAtContainer<TContainer,T>> getConfigAndContainer(ConfigKey<T> key) {
+        return getConfigImpl(key, false);
+    }
+
+    protected abstract TContainer getParentOfContainer(TContainer container);
+    
+    @Nullable protected final <T> ConfigKey<T> getKeyAtContainer(TContainer container, ConfigKey<T> queryKey) {
+        if (container==null) return null;
+        @SuppressWarnings("unchecked")
+        ConfigKey<T> candidate = (ConfigKey<T>) getKeyAtContainerImpl(container, queryKey);
+        return candidate;
+    }
+    
+    @Nullable protected abstract <T> ConfigKey<?> getKeyAtContainerImpl(@Nonnull TContainer container, ConfigKey<T> queryKey);
+
+    protected Maybe<Object> getRawValueAtContainer(TContainer container, ConfigKey<? extends Object> configKey) {
+        return ((BrooklynObjectInternal)container).config().getInternalConfigMap().getConfigLocalRaw(configKey);
+    }
+    protected Maybe<Object> resolveRawValueFromContainer(TContainer container, ConfigKey<?> key, Object value) {
```

Comment:
Worth some javadoc to say about the relationship between this method/impl and that in `ConfigInheritances` / `BasicConfigInheritance`. Will this respect the "reinheritable" etc, or is that the responsibility of the caller?